File: gr.demokritos.iit.graphmethods/src/gr/demokritos/iit/graphmethods/SimilarityCalculator.java
package gr.demokritos.iit.graphmethods;

import org.apache.spark.graphx.Graph;

/**
 *
 * @author JKLS
 */
public interface SimilarityCalculator {
    
    /**
     * 
     * @param g1 graph1
     * @param g2 graph2
     * @return similarity between two graphs
     */
    public Similarity getSimilarity(Graph g1, Graph g2);
    
}

File: gr.demokritos.iit.nGramGraphMethods/src/gr/demokritos/iit/nGramGraphMethods/SimilarityCalculator.scala
package gr.demokritos.iit.nGramGraphMethods

import org.apache.spark.graphx.Graph

/**
 * @author Kontopoulos Ioannis
 */
trait SimilarityCalculator {
  
  //@return similarity between two graphs
  def getSimilarity(g1: Graph[(String, Int), Int], g2: Graph[(String, Int), Int]): Similarity
  
}